shard/src/main/java/com/redhat/service/bridge/shard/controllers/IngressController.java

```diff
@@ -51,30 +51,39 @@
     @Inject
     KubernetesClient kubernetesClient;
 
-    void onEvent(@Observes ResourceEvent event) throws JsonProcessingException { // equivalent of ResourceEventSource for operator sdk
+    void onEvent(@Observes ResourceEvent event) { // equivalent of ResourceEventSource for operator sdk
         if (event.getSubject().equals(K8SBridgeConstants.BRIDGE_TYPE)) {
+
+            /*
+             * If the CRD is deleted, remove also all the other related resource
+             *
+             * If another dependent resource is deleted, the `reconcileExecutor` will catch the mismatch between the expected state and the current state.
+             * It will redeploy the resources so to reach the expected status at the end.
+             */
+            if (event.getAction().equals(Action.DELETED) && event.getResourceType().equals(KubernetesResourceType.CUSTOM_RESOURCE)) {
+                delete(event.getResourceId());
+                return;
+            }
```

Comment:
If we properly set [CR ownership and cascading](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#ownership-and-finalizers) this should become OOTB when moved to K8s: if the BridgeCR owns all the ProcessorCR, when it is removed, all dependents resources are automatically removed.
What about adding a TODO comment to track this so that when we move to K8s we properly check the configuration?

Reply:
Yes good point